eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/UnmodifiableBiMap.java

```diff
@@ -0,0 +1,1119 @@
+    }
+
+    @Override
+    public MutableBiMap<V, K> inverse()
+    {
+        return this.delegate.inverse().asUnmodifiable();
+    }
+
+    @Override
+    public MutableBiMap<V, K> flipUniqueValues()
+    {
+        return this.delegate.flipUniqueValues();
+    }
+
+    @Override
+    public V getIfAbsent(K key, Function0<? extends V> function)
+    {
+        return this.delegate.getIfAbsent(key, function);
+    }
+
+    @Override
+    public V getIfAbsentValue(K key, V value)
+    {
+        return this.delegate.getIfAbsentValue(key, value);
+    }
+
+    @Override
+    public <P> V getIfAbsentWith(K key, Function<? super P, ? extends V> function, P parameter)
+    {
+        return this.delegate.getIfAbsentWith(key, function, parameter);
+    }
+
+    @Override
+    public <A> A ifPresentApply(K key, Function<? super V, ? extends A> function)
+    {
+        return this.delegate.ifPresentApply(key, function);
+    }
+
+    @Override
+    public RichIterable<K> keysView()
+    {
+        return this.delegate.keysView();
+    }
+
+    @Override
+    public RichIterable<V> valuesView()
+    {
+        return this.delegate.valuesView();
+    }
+
+    @Override
+    public RichIterable<Pair<K, V>> keyValuesView()
+    {
+        return this.delegate.keyValuesView();
+    }
+
+    @Override
+    public ImmutableBiMap<K, V> toImmutable()
+    {
+        return this.delegate.toImmutable();
+    }
+
+    @Override
+    public MutableSetMultimap<V, K> flip()
+    {
+        return this.delegate.flip();
+    }
+
+    @Override
+    public V put(K key, V value)
+    {
+        throw new UnsupportedOperationException("Cannot call put() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public V remove(Object key)
+    {
+        throw new UnsupportedOperationException("Cannot call remove() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> m)
+    {
+        throw new UnsupportedOperationException("Cannot call putAll() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public void clear()
+    {
+        throw new UnsupportedOperationException("Cannot call clear() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public Set<K> keySet()
+    {
+        return Collections.unmodifiableSet(this.delegate.keySet());
+    }
+
+    @Override
+    public Collection<V> values()
+    {
+        return Collections.unmodifiableCollection(this.delegate.values());
+    }
+
+    @Override
+    public Set<Entry<K, V>> entrySet()
+    {
+        return Collections.unmodifiableMap(this.delegate).entrySet();
+    }
+
+    @Override
+    public V forcePut(K key, V value)
+    {
+        throw new UnsupportedOperationException("Cannot call forcePut() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public MutableBiMap<K, V> asSynchronized()
+    {
+        return this.delegate.asSynchronized();
```

Comment:
I think this should be something like SynchronizedBiMap.of(this), otherwise the synchronized wrapper becomes mutable. Check the unmodifiable wrappers of maps for comparison.